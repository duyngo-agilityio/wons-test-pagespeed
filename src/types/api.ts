export enum Method {
  Get = 'GET',
  Post = 'POST',
  Put = 'PUT',
  Patch = 'PATCH',
  Delete = 'DELETE',
}
