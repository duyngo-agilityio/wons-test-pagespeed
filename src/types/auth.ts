// Models
import { User } from '@/models';

export interface SignInFormData {
  identifier: string;
  password: string;
}

export interface AuthResponse {
  jwt: string;
  user: User | null;
  error: string;
}

export type TSignUpPayload = {
  fullName: string;
  username: string;
  email: string;
  password: string;
};

export type TSignUpResponse = { jwt: string; user: Omit<User, 'password'> };
