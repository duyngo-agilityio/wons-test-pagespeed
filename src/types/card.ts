export enum Level {
  Low = 'low',
  Medium = 'medium',
  High = 'high',
}
