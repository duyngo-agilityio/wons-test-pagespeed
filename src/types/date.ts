import { DateValue } from '@internationalized/date';

export interface DateRangeState {
  start: DateValue;
  end: DateValue;
}
