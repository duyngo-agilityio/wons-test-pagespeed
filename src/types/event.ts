// Models
import { ICalendarTask, IEvent, TUser } from '@/models';

// Types
import { StrapiModel } from './strapi';

export type TEventResponse = Omit<IEvent, 'users_permissions_users'> &
  ICalendarTask & {
    users_permissions_users: { data: StrapiModel<TUser>[] };
  };
