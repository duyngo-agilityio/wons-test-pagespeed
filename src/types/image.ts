export interface UploadImageResponse {
  downloadURL?: string;
  error?: string;
}
