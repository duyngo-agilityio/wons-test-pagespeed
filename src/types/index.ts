export * from './auth';
export * from './date';
export * from './params';
export * from './select';
export * from './strapi';
export * from './table';
export * from './toast';
export * from './invoice';
export * from './customer';
