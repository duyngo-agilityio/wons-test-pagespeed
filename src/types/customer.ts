// Models
import { ICustomer } from '@/models';

// Types
import { StrapiModel, StrapiResponse } from '@/types';

export type TCustomerDataResponse = StrapiModel<ICustomer>;

export type TCustomerListResponse = StrapiResponse<TCustomerDataResponse[]>;
