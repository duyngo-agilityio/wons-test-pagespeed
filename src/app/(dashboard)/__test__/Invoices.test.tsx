// Pages
import InvoiceListPage from '@/app/(dashboard)/invoices/page';

// Types
import { ISearchParams } from '@/types';

jest.mock('@/components', () => ({
  TableSkeleton: () => <div>Table Skeleton</div>,
}));

jest.mock('@/ui', () => ({
  InvoiceListActions: () => <div>InvoiceListActions</div>,
  InvoiceList: () => <div>InvoiceList</div>,
}));

jest.mock('@/layouts', () => ({
  DashBoardLayout: () => <div>DashBoardLayout</div>,
}));

describe('CustomerListPage render', () => {
  it('should render and match with snapshot', async () => {
    const { container } = testLibJestUtils.render(
      <InvoiceListPage searchParams={undefined as unknown as ISearchParams} />,
    );

    expect(container).toMatchSnapshot();
  });
});
