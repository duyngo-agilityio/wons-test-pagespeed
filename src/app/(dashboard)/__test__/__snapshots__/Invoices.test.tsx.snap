// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomerListPage render should render and match with snapshot 1`] = `
<div>
  <div
    data-overlay-container="true"
  >
    <script>
      !function(){try{var d=document.documentElement,c=d.classList;c.remove('light','dark');var e=localStorage.getItem('theme');if('system'===e||(!e&&false)){var t='(prefers-color-scheme: dark)',m=window.matchMedia(t);if(m.media!==t||m.matches){d.style.colorScheme = 'dark';c.add('dark')}else{d.style.colorScheme = 'light';c.add('light')}}else if(e){c.add(e|| '')}else{c.add('light')}if(e==='light'||e==='dark'||!e)d.style.colorScheme=e||'light'}catch(e){}}()
    </script>
    <main>
      <div>
        DashBoardLayout
      </div>
    </main>
  </div>
</div>
`;
